homework_9/quadratic_equation.py

```diff
@@ -0,0 +1,76 @@
+# Напишите следующие функции:
+# ○ Нахождение корней квадратного уравнения
+# ○ Генерация csv файла с тремя случайными числами в каждой строке. 100-1000 строк.
+# ○ Декоратор, запускающий функцию нахождения корней квадратного уравнения с каждой тройкой чисел из csv файла.
+# ○ Декоратор, сохраняющий переданные параметры и результаты работы функции в json файл.
+
+import csv
+import json
+import random
+from functools import wraps
+from pathlib import Path
+from typing import Callable
+
+
+def data_from_csv_wrap(file_name: str):
+    def deco(func: Callable):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            lst_solution = []
+            with open(file_name, 'r', encoding='utf-8') as csv_file:
+                reader = csv.reader(csv_file, quoting=csv.QUOTE_NONNUMERIC)
+                for row in reader:
+                    x_1, x_2 = func(row[0], row[1], row[2])
```

Comment:
Вы же не переводите в `int`, или `float`
```suggestion
                    x_1, x_2 = func(*map(int, row))
```